core/trino-main/src/test/java/io/trino/sql/planner/TestTableScanRedirectionWithPushdown.java

```diff
@@ -77,7 +77,8 @@
     private static final String TEST_TABLE = "test_table";
     private static final SchemaTableName SOURCE_TABLE = new SchemaTableName(TEST_SCHEMA, TEST_TABLE);
 
-    private static final Session MOCK_SESSION = testSessionBuilder().setCatalog(MOCK_CATALOG).setSchema(TEST_SCHEMA).build();
+    // Disable complex expression pushdown so CAST are not translated
+    private static final Session MOCK_SESSION = testSessionBuilder().setCatalog(MOCK_CATALOG).setSchema(TEST_SCHEMA).setSystemProperty("complex_expression_pushdown", "false").build();
```

Comment:
i don't think we want this.

- complex_expression_pushdown disables more than just CASTs
- complex_expression_pushdown=true is the default, and we want to first and foremost rigorously test the default setup

Reply:
complex_expression_pushdown breaks this test and I don't know how to correctly fix it

Reply:
let's revert here and iterate from there

Reply:
@raunaqmorarka can you help with that test?

Reply:
I think you need to modify the MockConnector setup in `createLocalQueryRunner` to not pushdown CAST. I'm guessing `mockApplyProjection` needs to be changed for that.

Reply:
I got it to work at https://github.com/raunaqmorarka/presto/commit/2316e60e746f24f12f50c52a1ac648462f8a55ae